### DISCUSSION TITLE
link-layer - building blocks

### DISCUSSION THREAD

**Discussion by ivanhofer** [pipeli18ne/RFC#5] [Feedback]
Does the concept of building blocks makes sense? Is something unclear?

**Comment by osdiab** [comment]
Makes sense, one quick feedback - the `dictionaries` part says its the "files" that the translations are stored in, though for certain use cases i18n data needs to be stored in databases instead. Is that out of scope of this?

**Reply by ivanhofer** [reply]
The idea of `pipeli18ne` is it to create a flexible approach that can be extended in multiple ways. A database would probably just be what I currently have described as `dictionary`. If we take the config example from [here](https://github.com/pipeli18ne/RFC/blob/main/01-link-layer.md#the-i18n-pipeline), then you would probably replace `@pipeli18ne/dictionary-json` with `@pipeli18ne/dictionary-postgresql` or `@pipeli18ne/dictionary-mongodb`.

Thanks for the feedback, I'll add a note to the RFC.
